crates/stages/src/test_utils/test_db.rs

```diff
@@ -183,4 +183,44 @@ impl TestTransaction {
             Ok(())
         })
     }
+
+    /// Insert ordered collection of [SealedBlock] into corresponding tables.
+    /// Superset functionality of [TestTransaction::insert_headers].
+    pub(crate) fn insert_blocks<'a, I>(
+        &self,
+        blocks: I,
+        tx_offset: Option<u64>,
+    ) -> Result<(), DbError>
+    where
+        I: Iterator<Item = &'a SealedBlock>,
+    {
+        self.commit(|tx| {
+            let mut current_tx_id = tx_offset.unwrap_or_default();
+            let blocks = blocks.collect::<Vec<_>>();
+
+            for block in blocks {
+                let key: BlockNumHash = block.num_hash().into();
+
+                // Insert into header tables.
+                tx.put::<tables::CanonicalHeaders>(block.number, block.hash())?;
+                tx.put::<tables::HeaderNumbers>(block.hash(), block.number)?;
+                tx.put::<tables::Headers>(key, block.header.clone().unseal())?;
+
+                // Insert into body tables.
+                tx.put::<tables::BlockBodies>(
+                    key,
+                    StoredBlockBody {
+                        start_tx_id: current_tx_id,
+                        tx_count: block.body.len() as u64,
+                    },
+                )?;
+                for body_tx in block.body.clone() {
```

Comment:
is this clone required? seems expensive

Reply:
yeah, because we are iterating over references